src/visitor/import-export.js

```diff
@@ -2,13 +2,16 @@ import MagicString from "../magic-string.js"
 import NullObject from "../null-object.js"
 import Visitor from "../visitor.js"
 
+import encodeId from "../util/encode-id.js"
 import getNamesFromPattern from "../parse/get-names-from-pattern.js"
 import toStringLiteral from "../util/to-string-literal.js"
 
 const codeOfCR = "\r".charCodeAt(0)
 
 const { keys } = Object
 
+const functionPrefixRe = /^\s*(?:async\s*)?function(?:\s*\*)?/
```

Comment:
Is there a AST node way to detect the async bit that doesn't require a source sniff?

Reply:
Unfortunately not, that would probably only be exposed by a Concrete Syntax Tree.

Reply:
There's a [`pp.isAsyncFunction`](https://github.com/ternjs/acorn/blob/0a3fb09fbc596bcaa4ff6201a0f5b743f96953f1/src/statement.js#L52-L62) acorn method. Does that help?

Reply:
No, it only checks if the function is async.

That regex is used to know where the `async function` or `function *` part ends, and the parser doesn't expose this information.

Reply:
Ah ok cool. I can work with that. Thanks for the clarification!